src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -746,5 +752,40 @@ public static ScreenshotEvent onScreenshot(BufferedImage image, File screenshotF
         MinecraftForge.EVENT_BUS.post(event);
         return event;
     }
+    
+    public static TransformType currentTransformType;
+    
+    public static List<List<BakedQuad>> sortQuadsByIndex(IBakedModel model)
+    {
+        List<List<BakedQuad>> layers = new ArrayList();
+        List<BakedQuad> quads = new ArrayList();
+        
+        for (EnumFacing dir : EnumFacing.values())
+        {
+            quads.addAll(model.getQuads(null, dir, 0));
+        }
+        quads.addAll(model.getQuads(null, null, 0));
+        
+        for (BakedQuad quad : quads)
+        {
+            int index = quad.hasTintIndex() ? quad.getTintIndex() : 0;
+            try { layers.get(index); } catch (IndexOutOfBoundsException e) { layers.add(index, new ArrayList()); }
```

Comment:
Wouldn't it be easier just to check if `index >= layers.size()` and add a new `ArrayList` to `layers` based on that?